carbon-design-system/carbon
# fix(styles): update notification to new token format

Part of #9358 

Our component token strategy currently requires teams to pass in the component tokens to the `theme` mixin directly in order for component tokens to work across themes. This PR updates `notification` so that its component tokens work in different themes and introduces a way to automatically register component tokens in the theme.

#### Changelog

**New**

- Add support for local component tokens to `themes` package
- Add support for getting a CSS Custom property for a component token

**Changed**

- Update eslint config to ignore `*.stories.js` files for prop type warnings
- Update storybook to use the correct background token
- Update notification story to include playground for toast
- Move all notification style tests into one test file
- Update notification to new token format, automatically register tokens

**Removed**

#### Testing / Reviewing

- Visit the notification playground story
- Turn on `lowContrast` in controls
- Switch between themes
- Verify that the background of the notification changes and the text remains readable
